unstructured_inference/models/table_postprocess.py

```diff
@@ -92,7 +95,8 @@ def refine_rows(rows, tokens, score_threshold):
     else:
         rows = nms(rows, match_criteria="object2_overlap", match_threshold=0.5, keep_higher=True)
     if len(rows) > 1:
-        rows = sort_objects_top_to_bottom(rows)
+        # Use a stable key to ensure deterministic output for equivalent rectangles
+        rows.sort(key=lambda k: (k["bbox"][1] + k["bbox"][3], k["bbox"][0] + k["bbox"][2]))
```

Comment:
sort_objects_top_to_bottom 
This is differrent as its considering X cordinate as well.